test/Datum.ts

```diff
@@ -186,6 +186,47 @@ describe('Datum', () => {
     );
   });
 
+  /**
+   * 1. Identity: `A.ap(A.of(a => a), fa) = fa`
+   * 2. Homomorphism: `A.ap(A.of(ab), A.of(a)) = A.of(ab(a))`
+   * 3. Interchange: `A.ap(fab, A.of(a)) = A.ap(A.of(ab => ab(a)), fab)`
+   */
+   it('ap', () => {
+    const f = (n: number) => n * 2;
+
+    assert.deepStrictEqual(D.Apply.ap(D.initial, D.initial), D.initial);
+    assert.deepStrictEqual(D.Apply.ap(D.initial, D.pending), D.initial);
+    assert.deepStrictEqual(D.Apply.ap(D.initial, D.refresh(1)), D.initial);
+    assert.deepStrictEqual(D.Apply.ap(D.initial, D.replete(1)), D.initial);
+
+    assert.deepStrictEqual(D.Apply.ap(D.pending, D.initial), D.pending);
```

Comment:
Note the mega instance above equals `D.initial`

https://github.com/nullpub/datum/pull/17/files#diff-c93edea33a3b97edd6aeec7ec663842689e3d415c918fcdd04c98192cae05a0eL161

The new `ap` instance short circuits on "no value", like chain must.